RBI-11: Create GET request for Categories

FarcasDavid opened the pull request:
- Created a model for Meals.
- Created a class called CategoriesService which inherits BaseURLService.
- Created methods for fetching all the categories and the meals by category.

Commit:
RBI-11:
- Created a model for Meals.
- Created a class called CategoriesService which inherits BaseURLService.
- Created methods for fetching all the categories and the meals by category.

edwardchristian97 commented on RecipeBook/Networking/CategoriesService.swift:
```diff
@@ -0,0 +1,32 @@
+
+class CategoriesService: BaseURLService {
+
+    var categoriesURL: String {
```
Can be private

edwardchristian97 commented on RecipeBook/Networking/CategoriesService.swift:
```diff
@@ -0,0 +1,32 @@
+            case .success(let data):
+                do {
+                    let json = try JSONDecoder().decode(RecipeMealModel.self, from: data)
+                    print(json.categories)
```
Remove print

edwardchristian97 commented on RecipeBook/Networking/CategoriesService.swift:
```diff
@@ -0,0 +1,32 @@
+                    let json = try JSONDecoder().decode(RecipeMealModel.self, from: data)
+                    print(json.categories)
+                } catch {
+                    print(error)
```
Remove print

edwardchristian97 commented on RecipeBook/Networking/MealsService.swift:
```diff
@@ -0,0 +1,32 @@
+            case .success(let data):
+                do {
+                    let json = try JSONDecoder().decode(MealsModel.self, from: data)
+                    print(json.meals)
```
Remove print

edwardchristian97 commented on RecipeBook/Networking/MealsService.swift:
```diff
@@ -0,0 +1,32 @@
+        return baseURL + "filter.php?c="
+    }
+
+    func fetchMealByCategory() {
```
Add a paramater which represents the category and append it to the URL

Commit:
RBI-11:
- Code refractoring
- Added completionHandler to fetchMealsByCategory & fetchAllCategories

Commit:
RBI-11:
- Made categoriesURL private

Commit:
no message
